src/middle/Fun_kind.ml

```diff
@@ -1,11 +1,29 @@
 open Core_kernel
 
+type 'a suffix = FnPure | FnRng | FnLpdf of 'a | FnTarget
+[@@deriving compare, sexp, hash, map]
+
 type t =
-  | StanLib of string
+  | StanLib of string * bool suffix
   | CompilerInternal of Internal_fun.t
-  | UserDefined of string
+  | UserDefined of string * bool suffix
```

Comment:
This looks awesome. Maybe you could add a comment saying what the bool means (normalization it looks like), since it doesn't have a name? A little comment or a link to the docs at the definition of the `suffix` variants might be helpful too.

Reply:
I renamed the type parameter on `suffix` to `'propto`. Does it need more docs?
The only reason it's even parametrized is because I want to use the same suffixes for function definitions and those don't make a distinction between normalized and unnormalized lpdfs.

Reply:
I do think a sentence or two would improve clarity, but this is good enough for me